### PR TITLE
[enhancement](compaction) Optimize calculating level size of input rowset in SizeBasedCumulativeCompactionPolicy

### PR DESCRIPTION
**Pull request by platoneko**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 36.8 seconds
 load time: 474 seconds
 storage size: 17122903688 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230104200921_clickbench_pr_73810.html

File: be/src/olap/cumulative_compaction_policy.cpp [hunk 1]
```diff
@@ -335,13 +325,12 @@ int SizeBasedCumulativeCompactionPolicy::pick_input_rowsets(
...
-int SizeBasedCumulativeCompactionPolicy::_level_size(const int64_t size) {
-    for (auto& i : _levels) {
-        if (size >= i) {
-            return i;
-        }
-    }
-    return 0;
+int64_t SizeBasedCumulativeCompactionPolicy::_level_size(const int64_t size) {
+    if (size < 1024) return 0;
+    int64_t max_level = (int64_t)1
```
Comment (yixiutt) [inline be/src/olap/cumulative_compaction_policy.cpp:330]: add some note here, it's hard to understand

Comment (platoneko) [reply be/src/olap/cumulative_compaction_policy.cpp:330]: Calculate the first 2 power number <= the given size.

Comment (platoneko) [reply be/src/olap/cumulative_compaction_policy.cpp:332]: done

**Review by dataroaring — DISMISSED** [review]
LGTM

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**